File: src/main/java/cz/vhromada/catalog/mapper/MusicMapper.java
package cz.vhromada.catalog.mapper;

import cz.vhromada.catalog.entity.Music;

import org.mapstruct.Mapper;
import org.mapstruct.Mapping;

/**
 * An interface represents mapper for music.
 *
 * @author Vladimir Hromada
 */
@Mapper
public interface MusicMapper {

    /**
     * Maps entity music to domain music.
     *
     * @param source entity music
     * @return mapped domain music
     */
    @Mapping(target = "songs", ignore = true)
    cz.vhromada.catalog.domain.Music map(Music source);

    /**
     * Maps domain music to entity music.
     *
     * @param source domain music
     * @return mapped entity music
     */
    Music mapBack(cz.vhromada.catalog.domain.Music source);

}

File: src/main/kotlin/cz/vhromada/catalog/mapper/MusicMapper.kt
package cz.vhromada.catalog.mapper

import cz.vhromada.catalog.entity.Music
import cz.vhromada.common.mapper.Mapper
import org.springframework.stereotype.Component

/**
 * A class represents mapper for music.
 *
 * @author Vladimir Hromada
 */
@Component("musicMapper")
class MusicMapper : Mapper<Music, cz.vhromada.catalog.domain.Music> {

    override fun map(source: Music): cz.vhromada.catalog.domain.Music {
        return cz.vhromada.catalog.domain.Music(
                id = source.id,
                name = source.name!!,
                wikiEn = source.wikiEn,
                wikiCz = source.wikiCz,
                mediaCount = source.mediaCount!!,
                note = source.note,
                position = source.position,
                songs = emptyList())
    }

    override fun mapBack(source: cz.vhromada.catalog.domain.Music): Music {
        return Music(
                id = source.id,
                name = source.name,
                wikiEn = source.wikiEn,
                wikiCz = source.wikiCz,
                mediaCount = source.mediaCount,
                note = source.note,
                position = source.position)
    }

}
